### ISSUE TITLE
Improve pairing with Mi Band

### ISSUE THREAD

**Issue by cpfeiffer** [Freeyourgadget/Gadgetbridge#180]
The pairing procedure needs improvement. I just used it with a previously "unknown" Mi 1A device and the following happened:
- it was successfully paired at the bluetooth level
- the progress kept on spinning
- the band never started to blink the LEDs

The reason for the procedure not working was probably that we land in "initializing" state too early and then nothing happens. I noticed this when I aborted the pairing activity. I then long-pressed the device item and chose "Disconnect". Then pressed the device item et voilà, the LEDs started to blink and the application level bonding succeeded.

I think we need to change the initialization procedure during the initial bonding to perform less.

**Comment by cpfeiffer** [comment]
Also see #165

**Comment by bigretromike** [comment]
I think that this is the source of this problem: https://github.com/Freeyourgadget/Gadgetbridge/issues/249 what you describe look really familiar, sometimes it stuck on inicialization and sometimes pair correctly, like there is some kind of lag that make initialization to long/short.

**Comment by cpfeiffer** [comment]
It's true that #249 could also influence the initial pairing, but the above mentioned scenario was solved differently. Now we register for notifications *early* so that we get a notification when authentication is required (blinking LEDs).

There still seems to be some timing problems, that's why this issue is still open.

**Comment by bigretromike** [comment]
#249 was resolve which is great, but band still like to sometimes disconnect from GB

**Comment by cpfeiffer** [comment]
That's true, but pleas eopen a separate ticket for that. Unless we find a way to reproduce that, it's not gonna be fixed soon, btw, since this can have many different causes. 

Two things we could do: 
1. add some kind of watchdog, that repeatedly attempts to reconnect (which would cost battery when phone and band are not at the same place)
2. let the bt stack on the phone do this again (we disabled this some time ago, but we might enable it again now that #249 is fixed

OTOH it really is not that necessary because we auto-connect to the band whenever a notification comes in.

**Comment by bigretromike** [comment]
oh, didn't know its auto-connect on notification- good to know, I only see point to be pair all time when doing training. other way we can live with autoconnect on notification
